### PR TITLE
add getting-started guide

### PR DESCRIPTION
**Pull request by josefaidt**
#### Description of changes:

- adds getting-started guide
- adds NodePackageManager context, hook
- adds NodePackageMangerTabs component
- adds NodePackageManagerInlineCodeBlock component for use in MDX pages when specifying "npm"/package manager commands (this is reactive to chosen package manager state)
- adds MDXCode types

Nice to have but not done yet:

- support for markdown fenced code block "title" (i.e. filename)
- support for markdown fenced code block line highlighting (via `{#}` like `{6}`)

Pending work/todo:

- ~~console images and alt text @swaminator~~

https://github.com/aws-amplify/docs/assets/5033303/86908dcc-5e3c-43ab-a254-eb2bd10b3645

https://github.com/aws-amplify/docs/assets/5033303/4cd1b924-7e25-41c5-b446-682ab76e561d

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [x] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/start/index.mdx [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+In order to share your app with others, Amplify will need to connect to a git repo as the source of truth for your project.
+
+### Create a remote git repository
+
+If you already have your project remotely hosted in a git repository, you can skip this step. Otherwise, navigate to your preferred git provider, and add your project to a new repo.
+
+Amplify supports the following git providers:
+
+- Github: `gh repo create`
```
Comment (josefaidt) [inline src/pages/gen2/start/index.mdx:223]:
```suggestion
- GitHub: `gh repo create`
```

File: src/pages/gen2/start/index.mdx [hunk 2]
```diff
@@ -0,0 +1,237 @@
+export const meta = {
+  title: 'Getting Started',
+  description: 'Learn how to get started with AWS Amplify (Gen 2)'
+};
+
```
Comment (heatheramz) [inline src/pages/gen2/start/index.mdx:5]: Recommend we add a short preview of what follows with an estimated time to complete end to end. Something like - "This *Getting Started* guide walks you through the setting up your project and personal cloud sandbox, configuring authentication and data, and deploying and hosting your app. If you are starting a new project this walk through will take you about XX minutes/hours to complete."

File: src/pages/gen2/start/index.mdx [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema
+});
+```
+
+## Authentication
+
+Each of the `resource.ts` files is where you _define_ the corresponding backend resource:
```
Comment (heatheramz) [inline src/pages/gen2/start/index.mdx:71]: May add a comment within this line on what type of auth is defined here - this is showing set up with email/password auth - correct?

For example, "Each of the `resource.tx` files is where you _define_ the corresponding backend resource. Here we are adding authentication using email.

File: src/pages/gen2/start/index.mdx [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+  return <Component {...pageProps} />;
+}
+
+export default withAuthenticator(App);
+```
+
+Run your application with `npm run dev` and navigate to `http://localhost:3000`. You should now see the authenticator, which is already configured and ready for your first sign-up!
+
+## Data
+
```
Comment (heatheramz) [inline src/pages/gen2/start/index.mdx:129]: Should add small text markers to transition the walk through / steps. Can be simple but for example, "Next you will want to set up your data resources to to save and read data to and from the cloud."

File: src/pages/gen2/start/index.mdx [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+### Connect repo to Amplify Console
+
+![d](/images/gen2/console1.jpg)
+
+### Manage your app
+
+![d](/images/gen2/console2.png)
+
+![d](/images/gen2/console3.png)
```
Comment (heatheramz) [inline src/pages/gen2/start/index.mdx:237]: Would a couple more sentences to call out what you want them to see on the images above. Also - are we going to add some recommended next steps at the end?

**Review by heatheramz — COMMENTED** [review]
Adding some initial comments on the content piece for @swaminator

File: src/components/NodePackageManagerTabs/NodePackageManagerTabs.tsx
Comment (timngyn) [inline src/components/NodePackageManagerTabs/NodePackageManagerTabs.tsx:1]: Hi @josefaidt, do you have a screenshot of what this component looks like?

Is this just so the site remembers what package manager a visitor uses?

Comment (josefaidt) [reply src/components/NodePackageManagerTabs/NodePackageManagerTabs.tsx:1]: Hey! :wave: I added a video of the component in the PR description. I started to looking at offloading the choice to `localStorage`, but this is mostly to keep the choice in sync throughout the document (and across pages in the same session)

File: src/pages/gen2/start/index.mdx (hunk 3 above)
Comment (josefaidt) [reply src/pages/gen2/start/index.mdx]: Added more of an intro here

File: src/pages/gen2/start/index.mdx (hunk 4 above)
Comment (josefaidt) [reply src/pages/gen2/start/index.mdx:152]: added more of an intro here

File: src/pages/gen2/start/index.mdx (hunk 5 above)
Comment (josefaidt) [reply src/pages/gen2/start/index.mdx]: @swaminator , thoughts on next steps here?

File: src/pages/gen2/start/index.mdx (hunk 2 above)
Comment (josefaidt) [reply src/pages/gen2/start/index.mdx:6]: Added the description and time estimate

**Comment by josefaidt** [comment]
added a quick effort to offload package manager choice to local storage to "remember" across visits, and accepts a `pm` query parameter to load from URL state in the event the visitor is coming from an article that references a specific workflow. Query parameter is not yet added to URL on state change

```tsx
useEffect(() => {
  // load "saved" package manager choice
  if (typeof window !== 'undefined' && window?.localStorage) {
    const saved = localStorage.getItem(LOCAL_STORAGE.NODE_PACKAGE_MANAGER);
    if (saved && NODE_PACKAGE_MANAGERS[saved]) {
      dispatch({ type: 'update', value: saved as NodePackageManager });
    }
  }
}, []);

useEffect(() => {
  // save package manager choice on change
  if (typeof window !== 'undefined' && window?.localStorage) {
    localStorage.setItem(LOCAL_STORAGE.NODE_PACKAGE_MANAGER, state);
  }
}, [state]);

useEffect(() => {
  // since this runs last, we can load package manager choice from URL (e.g. visiting from an article)
  if (typeof window !== 'undefined' && window.location.search) {
    const searchParams = new URLSearchParams(window.location.search);
    const packageManager = searchParams.get(
      URL_SEARCH_PARAMS.NODE_PACKAGE_MANAGER
    );
    if (packageManager && NODE_PACKAGE_MANAGERS[packageManager]) {
      dispatch({
        type: 'update',
        value: packageManager as NodePackageManager
      });
    }
  }
}, []);
```

**Comment by josefaidt** [comment]
thinking about adding this to URL state on change and I have a few follow-up thoughts on next steps to increase discoverability of sharing with pm state:

- we can change the name of the query param
- we can build this so that each tab change adds the choice to URL state (i.e. `?pm=pnpm`)
	- this seems a bit aggressive for sharing general workflows, but is also highly visible
	- going this route makes the localStorage capability redundant
- we can build a "share" functionality that allows you to share the page with all relevant tab state (this achieves the use case of sharing a doc referencing a specific flow -- e.g. "getting started with pnpm and Next.js with app router")

**Comment by josefaidt** [comment]
@heatheramz thoughts on moving this from `/gen2/start` to `/gen2/getting-started` to align with the doc name?

**Comment by josefaidt** [comment]
closing as content has been merged and components will be revisited

**Pull request closed without merging**